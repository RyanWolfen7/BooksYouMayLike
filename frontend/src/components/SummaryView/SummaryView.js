import React from 'react'
import { SummaryViewStyles } from '../../styles/SummaryViewStyles'
import { Card, CardContent, CardHeader, Typography, GridList, GridListTile } from '@material-ui/core'

const SummaryView = props => {
    const classes = SummaryViewStyles()
    const book = props.book

    return (
        <Card className={classes.info}>
            <CardHeader
                className={classes.header}
                title={book.name}
                subheader={<>
                    <span>by: {book.writer}</span>
                    <div>published: {new Date(book.date_published * 1000).toDateString()}</div>
                </>}
            />
            <CardContent >
                <Typography className={classes.content} align={'center'} variant={'subtitle1'}> Summary </Typography>
                <GridList col={1} cellHeight={'auto'} className={classes.gridList}>
                    <Typography className={classes.summary} variant={'body1'}> {book.summary} </Typography>
                </GridList>
            </CardContent>
        </Card>
    )
}

export default SummaryView