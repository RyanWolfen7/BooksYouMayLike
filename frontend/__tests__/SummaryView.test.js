import React from 'react'
import { shallow } from 'enzyme'
import SummaryView from '../src/components/SummaryView/SummaryView'

let view

beforeAll(() => {
    view = shallow(<SummaryView />)
})

test('view to match snapshot', () => {
    expect(view).toMatchSnapshot()
})