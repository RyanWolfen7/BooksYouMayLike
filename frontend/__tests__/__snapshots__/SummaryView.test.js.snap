// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`view to match snapshot 1`] = `
<WithStyles(ForwardRef(Card))
  className="makeStyles-info-2"
/>
`;
